scipy/sparse/_construct.py

```diff
@@ -1250,15 +1244,16 @@ def data_sampler(size):
     return vals, ind
 
 
+@_transition_to_rng("random_state", position_num=5)
 def random(m, n, density=0.01, format='coo', dtype=None,
-           random_state=None, data_rvs=None):
+           rng=None, data_rvs=None):
     """Generate a sparse matrix of the given shape and density with randomly
     distributed values.
 
     .. warning::
 
         Since numpy 1.17, passing a ``np.random.Generator`` (e.g.
-        ``np.random.default_rng``) for ``random_state`` will lead to much
+        ``np.random.default_rng``) for ``rng`` will lead to much
```

Comment:
Maybe we should just delete this warning since it now reads a little weirdly and it refers to a pretty old numpy version.

Reply:
I agree that we don't need to be warning about changes from so long ago. It's not really related to this PR, but if there is more, I can include that with the other changes.

Reply:
I would argue it is somewhat related in that the intent of the message is now slightly confusing. If my interpretation is correct, it previously told users that if they changed the type of `random_state` from `RandomState` to  `Generator` they should expect a faster runtime however with the `rng` argument it is not possible to invoke the "slow" path

Reply:
I see. I'll address that with any other changes.